src/twisted/internet/defer.py

```diff
@@ -1505,14 +1550,13 @@ class _CancellationStatus:
     waitingOn = attr.ib(default=None)  # type: Optional[Deferred[object]]
 
 
-_InlineCallbacksGenerator = Union[
-    CoroutineType, Generator[Deferred[object], object, None]
-]
-
-
 @_extraneous
 def _inlineCallbacks(
-    result: object, gen: _InlineCallbacksGenerator, status: _CancellationStatus
+    result: object,
+    gen: Union[
+        Generator[Deferred[_T], object, None], Coroutine[Deferred[_T], object, None]
```

Comment:
Is None correct for the generator/coroutine return type for these? I think it should be Optional[object] since this handles pulling the value off of StopIteration(x) when the generator returns x.

And I don't think YieldType is necessarily a Deferred when gen is a Generator.

I think the same concerns apply to the original PR fwiw.